### PR TITLE
Add fluct to amp-ad support

### PR DESCRIPTION
**Pull request by saxsir**
Hi, we want to add fluct into the supported SSP for amp-ad.

We have signed Google Corporate CLA.

Corporation name: fluct, Inc.

Thank you very much !

### CODE REVIEW

**Comment by saxsir** [comment]
@ampprojectbot retry!

**Comment by saxsir** [comment]
@lannka  Hi, now CI test has all passed, I would like you to review our code.

**Comment by saxsir** [comment]
@zhouyx Could you review this PR?

File: examples/ads.amp.html [hunk 1]
```diff
@@ -641,6 +642,15 @@ <h2 id="flite">Flite</h2>
       data-mixins="">
   </amp-ad>
 
+  <h2 id="fluct">fluct</h2>
+  <amp-ad width="300" height="250"
+      type="fluct"
+      data-g="1000067784"
+      data-u="1000101409">
+    <div placeholder></div>
```
Comment (zhouyx) [inline examples/ads.amp.html]: with default placeholder supported. no need to add placeholder and fallback here.

File: extensions/amp-ad/amp-ad.md [hunk 2]
```diff
@@ -121,6 +121,7 @@ resources in AMP. It requires a `type` argument that select what ad network is d
 - [E-Planning](../../ads/eplanning.md)
 - [Ezoic](../../ads/ezoic.md)
 - [FlexOneELEPHANT](../../ads/f1e.md)
+- [fluct](../../ads/fluct.md)
```
Comment (zhouyx) [inline extensions/amp-ad/amp-ad.md]: `Fluct` please

File: ads/fluct.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    data-g="{GROUP-ID}"
+    data-u="{UNIT-ID}">
+</amp-ad>
+```
+
+## Configuration
+
+For more information, please [contact us](https://corp.fluct.jp/en/contact.php).
+
+Supported parameters:
```
Comment (zhouyx) [inline ads/fluct.md]: From your code, `data-g` and `data-` should be mandatory. please indicate they're required in the doc.

File: ads/fluct.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+/* global adingoFluct: false */
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function fluct(global, data) {
+  validateData(data, ['g', 'u']);
+  writeScript(global, `https://cdn-fluct.sh.adingo.jp/f.js?G=${data['g']}`, function() {
```
Comment (zhouyx) [inline ads/fluct.js]: use `encodeURIComponent` to wrap `data['g']` please

File: ads/_config.js [hunk 5]
```diff
@@ -266,6 +266,14 @@ export const adConfig = {
 
   flite: {},
 
+  fluct: {
+    preconnect: [
+      'https://cdn-fluct.sh.adingo.jp',
+      'https://s.sh.adingo.jp',
+      'https://i.adingo.jp',
+    ],
```
Comment (zhouyx) [inline ads/_config.js]: Please consider implementing the render-start and no-content-available APIs ([see Available APIs](https://github.com/voyagegroup/amphtml/blob/f5336dd4b61b7e80409a6690b120ec6af6d34d36/ads/README.md#available-apis)), which helps AMP to provide user a much better ad loading experience.

Please let me know if there's any questions on it.

File: ads/fluct.js (hunk 4 above)
Comment (zhouyx) [inline ads/fluct.js]: please follow the max 80 characters rule.

File: extensions/amp-ad/amp-ad.md (hunk 2 above)
Comment (saxsir) [inline extensions/amp-ad/amp-ad.md]: Our corporate name is "fluct"(not "Fluct").

https://corp.fluct.jp/en/

File: ads/_config.js (hunk 5 above)
Comment (saxsir) [inline ads/_config.js]: We are not supported API which kind of detect ad status(start rendring and no content) yet.

**Comment by saxsir** [comment]
@zhouyx

Thank you for your review.
We fixed some points you commented, please have a look.

**Comment by saxsir** [comment]
@ampprojectbot retry!

File: ads/fluct.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+/* global adingoFluct: false */
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function fluct(global, data) {
+  validateData(data, ['g', 'u']);
+  writeScript(global,
+    `https://cdn-fluct.sh.adingo.jp/f.js?G=${encodeURIComponent(data['g'])}`,
```
Comment (zhouyx) [inline ads/fluct.js]: extra indentation needed for code can't fall in one line. (need 2 more here and below).

Comment (saxsir) [inline ads/fluct.js]: @zhouyx Thx, added indent here.

**Comment by saxsir** [comment]
Fixed indent.

**Comment by zhouyx** [comment]
@saxsir LGTM. Please rebase to sync with master in order to make travis green.

**Comment by saxsir** [comment]
rebased.

@zhouyx Thank you very much!

**Comment by zhouyx** [comment]
Oops. @saxsir seems someone changes amp-ad.md as well. Please rebase again to fix the conflict, and then  it should be good to be merged.

**Comment by saxsir** [comment]
Oh... rebased. @zhouyx

**Comment by zhouyx** [comment]
@saxsir Thanks for the Pull Request. Merged.

**Pull request merged**